lib/board.rb

```diff
@@ -0,0 +1,63 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(cell)
+    get_cell = cell.to_i - 1
+    @cells[get_cell]
+  end
+
+  def full?
+    if @cells.include?(" ") then
+      return false
+    else
+      return true
+    end
+  end
+
+  def turn_count
+    counter = 0
+    @cells.each do |contents|
+      if contents != ' '
+        counter += 1
+      end
+    end
+    return counter
+  end
+
+  def taken?(location)
+    location = location.to_i - 1
+    !(@cells[location].nil? || @cells[location] == " ")
+  end
+
+  def valid_move?(location)
+    location = location.to_i
```

Comment:
Use 'and' operator (`&&`) to combine multiple logical expressions and utilize their implicit return value.
```
location.to_i.between?(1,9) && !taken(location)
```